File: HwangSettong/week_3/158p.py
# https://school.programmers.co.kr/learn/courses/30/lessons/12973?language=python3
def solution(s):
    stack=[]
    for i in s:
        if stack[-1:] == [i]:
            stack.pop()
        else:
            stack.append(i)
    return int(len(stack) == 0)

File: HwangSettong/week_3/158p.java
import java.util.Stack;
class Solution {
	public int solution(String s) {
		Stack<Character> stack = new Stack<Character>();
		for (int i = 0; i < s.length(); i++) {
			if (!stack.empty() && s.charAt(i) == stack.lastElement()) {
				stack.pop();
			} else {
				stack.push(s.charAt(i));
			}
		}
		if (stack.empty()) {
			return 1;
		}
		return 0;
	}
}
